src/lib/comms/sol-lwm2m.c

```diff
@@ -3085,6 +3163,127 @@ handle_create(struct sol_lwm2m_client *client,
     return SOL_COAP_RESPONSE_CODE_BAD_REQUEST;
 }
 
+static uint8_t
+handle_write(struct sol_lwm2m_client *client,
+    struct obj_ctx *obj_ctx, struct obj_instance *obj_instance,
+    uint16_t *path, uint16_t path_size,
+    struct sol_lwm2m_payload payload)
+{
+    int r;
+
+    if (client->is_bootstrapping) {
+        clear_bootstrap_ctx(client);
+    }
+
+    if (path_size < 2 && client->is_bootstrapping) {
+        //Bootstrap Write on Object (e.g.: PUT /1)
+        //In this case the payload is composed of <multiple TLVs> of type OBJECT_INSTANCE,
+        // <each one containing multiple TLVs> of type MULTIPLE_RESOURCES or RESOURCE_WITH_VALUE
+        // and each TLV of type MULTIPLE_RESOURCES can contain multiple TLVs of type
+        // RESOURCE_INSTANCE
+        struct sol_lwm2m_tlv *instance_tlvs;
+        uint16_t i;
+
+        if (payload.type == SOL_LWM2M_CONTENT_TYPE_TLV) {
```

Comment:
If the content type is not TLV you're still returning `SOL_COAP_RESPONSE_CODE_CHANGED`, which is not true.
If the payload type is not TLV you should return an error. At least for now